addons/calendar/models/calendar_event.py

```diff
@@ -641,6 +659,33 @@ def _attendees_values(self, partner_commands):
         ]
         return attendee_commands
 
+    def _create_videocall_channel(self):
+        if self.recurrency:
+            # check if any of the events have videocall_channel_id, if not create one
+            event_with_channel = self.env['calendar.event'].search([
+                ('recurrence_id', '=', self.recurrence_id.id),
+                ('videocall_channel_id', '!=', False)
+            ], limit=1)
+            if event_with_channel:
+                self.videocall_channel_id = event_with_channel.videocall_channel_id
+            else:
+                self.videocall_channel_id = self._create_videocall_channel_id(self.name, self.partner_ids.ids)
+                self.videocall_channel_id.channel_change_description(self.recurrence_id.name)
+        else:
+            self.videocall_channel_id = self._create_videocall_channel_id(self.name, self.partner_ids.ids)
+            self.videocall_channel_id.channel_change_description(self.display_time)
+
+    def _create_videocall_channel_id(self, name, partner_ids):
+        videocall_channel_id = self.env['mail.channel'].create_group(partner_ids, default_display_mode='video_full_screen', name=name)
+        # if recurrent event, set channel to all other records of the same recurrency
+        if self.recurrency:
+            recurrent_events_without_channel = self.env['calendar.event'].search([
+                ('recurrence_id', '=', self.recurrence_id.id), ('videocall_channel_id', '=', False)
+            ])
+            recurrent_events_without_channel.videocall_channel_id = videocall_channel_id['id']
+        return videocall_channel_id['id']
```

Comment:
Do you have a specific need to return just then id rather than the record? We are better off working with records in internal code, and only access `.id` when absolutely necessary.

Reply:
@seb-odoo From what I understood, `create_group` returns the first dictionary of a list of dictionaries and not a record.

Reply:
Indeed `create_group` is a public method, so it returns what it does to be called by RPC (can't return record by RPC, only data). I think a decorator exists to determine what is returned by RPC and internal call, but I never used it and I'm not sure if I even like the idea of changing the return API on the fly.

Anyway what we should probably do is to make create_group private, make it return a record always, and make a public route to call the method and return the channel_info.

I guess you can ignore this comment then.